numba/interpreter.py

```diff
@@ -171,9 +216,71 @@ def _start_new_block(self, offset):
                 break
 
     def _end_current_block(self):
+        # Handle try block
+        if not self.current_block.is_terminated:
+            tryblk = self.dfainfo.active_try_block
+            if tryblk is not None:
+                self._insert_exception_check()
+        # Handle normal block cleanup
         self._remove_unused_temporaries()
         self._insert_outgoing_phis()
 
+    def _insert_try_block_begin(self):
+        """Insert IR-nodes to mark the start of a `try` block.
+        """
+        gv_fn = ir.Global(
+            "mark_try_block", eh.mark_try_block, loc=self.loc,
+        )
+        mark_try_name = 'mark_try_fn'
+        self.store(value=gv_fn, name=mark_try_name, redefine=True)
+        mark_try = ir.Expr.call(self.get(mark_try_name), (), (), loc=self.loc)
+        self.store(value=mark_try, name='$mark_try', redefine=True)
+
+    def _insert_try_block_end(self):
+        """Insert IR-nodes to mark the end of a `try` block.
+        """
+        gv_fn = ir.Global(
+            "end_try_block", eh.end_try_block, loc=self.loc,
+        )
+        end_try_name = 'end_try_fn'
+        self.store(value=gv_fn, name=end_try_name, redefine=True)
+        end_try = ir.Expr.call(self.get(end_try_name), (), (), loc=self.loc)
+        self.store(value=end_try, name='$end_try', redefine=True)
```

Comment:
not essential but these are sufficiently similar they could share most of the lines.